app/models/metasploit/credential/krb_enc_key.rb

```diff
@@ -0,0 +1,123 @@
+require 'json'
+
+# A {Metasploit::Credential::PasswordHash password hash} that cannot be replayed to authenticate to other services.
+# {#data} is a serialized ruby hash with the keys defined in {Metasploit::Credential::KrbEncKey::ACCEPTABLE_DATA_ATTRIBUTES}
+#
+# This class contains information relevant to a Kerberos EncryptionKey https://www.rfc-editor.org/rfc/rfc4120.html#section-5.2.9
+# which is used to encrypt/decrypt arbitrary Kerberos protocol message data - such as the AS-REP krbtgt ticket and enc-part.
+class Metasploit::Credential::KrbEncKey < Metasploit::Credential::PasswordHash
+  # TODO: What are the security rammifications of this if we leak user controlled input here
+  serialize :data, MetasploitDataModels::Base64Serializer.new(default: {})
```

Comment:
This causes issues with the metasploit-credential search logic - as everywhere assumes that this field will be a string. I think for MVP we'll have to switch this to using a custom string format instead. Unfortauntely it looks like JTR doesn't have a predefined string format for this - as the existing formats assume a kerberos response of some kind will also be present:

```
7500  Kerberos 5, etype 23, AS-REQ Pre-Auth $krb5pa$23$user$realm$salt$4e751db65422b2..etc...
13100 Kerberos 5, etype 23, TGS-REP *user$realm$test/spn*$63386d22d359fe42230..etc...
18200 Kerberos 5, etype 23, AS-REP user@domain.com:3e156ada591263b8aab0965f5aebd837$007497cb51b6c8116d6407a78...etc...
... etc...
```

So we'll have to define our own:

```
```